app/Controllers/entryController.php

```diff
@@ -102,6 +102,41 @@ public function readAction(): void {
 						break;
 					case 't':
 						$entryDAO->markReadTag($get, $id_max, FreshRSS_Context::$search, FreshRSS_Context::$state, $is_read);
+						//marking all entries in a tag as read can result in other tags also having all entries marked as read,
+						//so the next unread tag calculation is deferred by passing next_get = 'a' instead of the current get ID.
+						if ($next_get === 'a' && $is_read) {
+							$tagDAO = FreshRSS_Factory::createTagDao();
+							$found_tag = 0;
+							foreach ($tagDAO->listTags() ?: [] as $tag) {
```

Comment:
This adds an SQL query (times 2, a bit below), so let's see if we can find a cheaper way to achieve the same

Reply:
I saw it did that, but couldn't come up with an alternative. Effectively we need the list of tags with unread articles in it, starting with the current tag and warpping back to the beginning. The current tag won't show up in the list of unread since we just marked it all as read, so I had to get the entire list just to figure out the label order, then query each label to find which has unread.

Maybe if we had a way to get a 2-dimemsional array of tags and their unread counts I could do it with a single query, but I don't see any way to achieve the order-based search for the next unread than the looping thru the list to find a matching ID, and then continuing the (wrapping) loop looking for the next unread.

Reply:
Done.